model/src/main/kotlin/SemanticVersionRange.kt

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model
+
+import com.vdurmont.semver4j.Semver
+
+import java.time.LocalDate
+import java.time.Period
+
+/**
+ * A data class to represent a range of semantic versions.
+ *
+ * The bounds of the range are defined by [Semver] objects. For a version object or a string referencing a version it
+ * is possible to check whether it is contained in the range. Ranges can be created either by specifying the bounds
+ * or in relation to a given base version and a level or "granularity". In the latter case, the range contains the
+ * base version and all versions that differ from it only to a certain degree, e.g. in the patch level.
+ *
+ * This class also offers some support for [Calendar versioning][https://calver.org/] (CalVer), which is used by some
+ * tools ORT has to deal with. A CalVer is a semantic version, but the components of the version are derived from the
+ * date when this version was released. When using such versions, ranges can be constructed that span a specific
+ * period around a base version; so this would cover releases within a specific time frame.
+ */
+data class SemanticVersionRange(
```

Comment:
Are you aware of `com.vdurmont.semver4j.Requirement`, or the various comparison function of `Semver`, like `Semver.diff()`? They seem to serve a similar purpose as this new class.

Reply:
I had a look at these features, but they did not seem to do quite the thing that was needed. After having a second look, the `Requirement` class could be a replacement for specifying a minimum and maximum version; the user could rather provide an expression following [NPM versioning guides](https://github.com/npm/node-semver). This would, however, still not handle these delta checks and CalVers.

But do you agree in principle that we should support such kind of configuration as "a version that differs only in the patch level from the scanner used by ORT" or "a CalVer released in a specific time frame to the scanner used by ORT"?